common/v2/services/Store/Asset/AssetProvider.tsx

```diff
@@ -16,7 +21,27 @@ export const AssetProvider: React.FC = ({ children }) => {
 
   const state: IAssetContext = {
     assets,
-    createAssetWithID: model.createWithID
+    createAssetWithID: model.createWithID,
+    getAssetByUUID: (uuid) => getAssetByUUID(assets)(uuid),
+    addAssetsFromAPI: (newAssets) => {
+      const mappedAssets = Object.entries(newAssets).map(([uuid, asset]: [TUuid, StoreAsset]) => ({
+        ...asset,
+        uuid,
+        isCustom: false
+      }));
+
+      // make a copy of current assets array and merge it with assets received from API
+      const assetsCopy = clone(assets);
+      mappedAssets.forEach((asset) => {
+        const existing = getAssetByUUID(assetsCopy)(asset.uuid);
+        if (existing) {
+          Object.assign(existing, asset);
+        } else {
+          assetsCopy.push(asset);
+        }
+      });
+      model.updateAll(assetsCopy);
+    }
```

Comment:
I think we can simplify with:

```
const setIsCustom = a => ({ ...a, isCustom: false })
const mergeAssets = pipe(
  reduce((acc, a) => ({ ...acc, [a.uuid]: a }), {}), // Transform user custom assets into object
  mergeLeft(map(setIsCustom, api)),                   // UUID is unique so we can merge user and api assets
  toPairs,                                           // Equivalent of Object.entries -> [k, v]
  map(([uuid, a]) => ({ ...a, uuid }))               // We Need to add the uuid key to the api asset.
)

model.updateAll(mergeAssets(assets))

```

[Demo](https://ramdajs.com/repl/#?%0Aconst%20api%20%3D%20%7B%0A%20%20123%3A%20%7B%20name%3A%20%27DAI%27%20%7D%2C%0A%20%20124%3A%20%7B%20name%3A%20%27SAI%27%20%7D%2C%0A%7D%0A%0Aconst%20assets%20%3D%20%5B%0A%20%20%7B%20name%3A%20%27DAI%27%2C%20uuid%3A%20123%20%7D%2C%0A%20%20%7B%20name%3A%20%27Custom%27%2C%20uuid%3A%20223%20%7D%0A%5D%0A%0Aconst%20setIsCustom%20%3D%20a%20%3D%3E%20%28%7B%20...a%2C%20isCustom%3A%20false%20%7D%29%0Aconst%20mergeAssets%20%3D%20pipe%28%0A%20%20reduce%28%28acc%2C%20a%29%20%3D%3E%20%28%7B%20...acc%2C%20%5Ba.uuid%5D%3A%20a%20%7D%29%2C%20%7B%7D%29%2C%20%2F%2F%20Transform%20user%20custom%20assets%20into%20object%0A%20%20mergeLeft%28map%28setIsCustom%2C%20api%29%29%2C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20UUID%20is%20unique%20so%20we%20can%20merge%20user%20and%20api%20assets%0A%20%20toPairs%2C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20Equivalent%20of%20Object.entries%20-%3E%20%5Bk%2C%20v%5D%0A%20%20map%28%28%5Buuid%2C%20a%5D%29%20%3D%3E%20%28%7B%20...a%2C%20uuid%20%7D%29%29%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20We%20Need%20to%20add%20the%20uuid%20key%20to%20the%20api%20asset.%0A%29%0A%0AmergeAssets%28assets%29)

Reply:
Used your solution, thanks!